### PR TITLE
Update Dolog version

### PR DESCRIPTION
**Pull request by andersfischernielsen**
This PR updates the `Dolog` version to >= 4.0.0 in order to build on recent OCaml versions. 

`Log` has been moved to `Dolog.Log` as described in https://github.com/UnixJunkie/dolog/blob/master/src/example.ml

### CODE REVIEW

**Comment by IagoAbal** [comment]
Thanks @andersfischernielsen. It seems that this PR depends on commit c6826718426b0946242be, isn't it?

File: eba.opam [hunk 1]
```diff
@@ -1,4 +1,4 @@
-opam-version: "1.2"
```
Comment (IagoAbal) [inline eba.opam]: This change should probably be part of PR #3, in case you want to keep two separate PR. For me it's fine if you squash them.

**Comment by andersfischernielsen** [comment]
Not necessarily, as far as I know. 😄 As far as I can read on the Dolog package on https://opam.ocaml.org/packages/dolog/dolog.4.0.0/ it should be compatible with older OCaml versions. Though when I installed `eba`s dependencies on OCaml 4.07.0 I got Dolog 4.0.0. 

It might just default to the most recent version. In order to make sure it's at least on the newest version I added the line https://github.com/IagoAbal/eba/blob/c6826718426b0946242be374c6ce25e7a6cb28f1/eba.opam#L20 in c682671 just to make sure.

**Review by IagoAbal — COMMENTED** [review]
Similarly to PR #3, I would appreciate a lot if you can squash these commits together so that we have a single and clean commit.

As I said, feel welcome to merge both PR and put together a larger commit titled "Make EBA build with a modern OCaml ecosystem" or something.

Thanks!
